File: leetcode/src/main/java/leetcode/support/ListNode.java
package leetcode.support;

import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
public class ListNode {
  public int val;
  public ListNode next;

  public ListNode(int val) {
    this(val, null);
  }

  public static ListNode of(int... values) {
    ListNode head = new ListNode();
    ListNode it = head;
    for (int value : values) {
      it.next = new ListNode(value);
      it = it.next;
    }
    return head.next;
  }

  @Override
  public String toString() {
    return val + "->" + next;
  }
}

File: src/main/kotlin/leetcode/ListNode.kt
package leetcode

class ListNode(
    var `val`: Int,
) {
    var next: ListNode? = null

    companion object {
        fun of(vararg values: Int): ListNode? {
            val head = ListNode(-1)
            var tail: ListNode? = head
            values.forEach { value ->
                tail?.next = ListNode(value)
                tail = tail?.next
            }
            return head.next
        }
    }
}
